### PR TITLE
fix(sql): add RleDictionary encoding support for byte columns in the read_parquet function

### LINKED ISSUE #4972: QDB Unable to Read Parquet File
### To reproduce

1. go to https://numer.ai/data/v5.0
2. download train.parquet
3. `read_parquet('r839_v5_0_train.parquet')`
4. see `Error reading. Parquet file is likely corrupted`

### QuestDB version:

8.1.1

### OS, in case of Docker specify Docker and the Host OS:

windows 10

### File System, in case of Docker specify Host File System:

NTFS

### Full Name:

Super Richman

### Affiliation:

None

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [X] Yes, I have

### Additional context

qdb log
```
2024-09-23T09:28:03.514855Z I i.q.g.e.QueryProgress exe [id=13, sql=`read_parquet('r839_v5_0_train.parquet')`, principal=admin, cache=false, jit=false]
2024-09-23T09:28:03.598328Z E i.q.g.e.t.p.PartitionDecoder could not decode [fd=27487790695546, columnId=2, rowGroup=0, msg=error in decodeColumnChunk: failed to decode row group [column=feature_shaded_hallucinatory_dactylology, group=0]

Caused by:
    encoding not supported, physical type: Int32, encoding RleDictionary, logical type Some(Integer(Int8)), converted type: Some(Int8), column type Byte

Stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>]
2024-09-23T09:28:03.598966Z E i.q.g.e.QueryProgress err [id=13, sql=`read_parquet('r839_v5_0_train.parquet')`, principal=admin, cache=false, jit=false, time=84176800, msg=[-1] Error reading. Parquet file is likely corrupted, errno=-1, pos=0]
2024-09-23T09:28:03.599464Z E i.q.c.h.p.JsonQueryProcessorState [5583457485126] error [msg=`Error reading. Parquet file is likely corrupted`, errno=-1, q=`read_parquet('r839_v5_0_train.parquet')`]
```

### PR DESCRIPTION
**Pull request by eugenels**
This PR fixes #4972

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `ci`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`, `ui`, `wal`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ab99d38cc31023fe5f1c75fa5b0af3fc7733b060
</p>

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|parquet_read/decode.rs|15|15|100.00%|

**Review by ideoma — DISMISSED** [review]

File: core/src/test/resources/sqllogictest/data/parquet-testing/bug4972.parquet
Comment (ideoma) [inline core/src/test/resources/sqllogictest/data/parquet-testing/bug4972.parquet:1]: 1.99 MB file with 10 rows in it, can we make it smaller

Comment (eugenels) [reply core/src/test/resources/sqllogictest/data/parquet-testing/bug4972.parquet:1]: replaced with a ~915KB

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|parquet_read/decode.rs|15|15|100.00%|

**Pull request merged**